### PR TITLE
Deprecate `clickhouseOperator.useNodeSelector`

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Description
With https://github.com/PostHog/charts-clickhouse/pull/212 we added a new `clickhouse.nodeSelector` chart variable to customize the `nodeSelector` definition of the ClickHouse installation.

This PR removes the now deprecated `clickhouseOperator.useNodeSelector` chart value. Looking at the how `altinity/clickhouse-operator` works, I doubt this value in our repo had ever worked as expected. 

In any case, to keep any eventual retro compatibility `clickhouseOperator.useNodeSelector: true` can be modified as:
```
clickhouse:
  nodeSelector:
    clickhouse: "true"
```

## Type of change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## How has this been tested?
CI is passing

## Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by tiina303** [comment]
LGTM thanks

**Comment by tiina303** [comment]
could you also add the new nodeSelector to values with a comment explaining it, so it's more easily discoverable for us & users

**Comment by tiina303** [comment]
related https://github.com/PostHog/posthog.com/pull/2671

**Comment by fuziontech** [comment]
I believe Vendasta is using this so I'll check with them to make sure they have new settings in place.

**Comment by tiina303** [comment]
can we move forward with this? Yesterday the useNodeSelector setting didn't work on a self-hosted install and rather than investigate further I think we should just move ahead and deprecate as has been the plan anyway.

**Comment by guidoiaquinti** [comment]
> can we move forward with this? Yesterday the useNodeSelector setting didn't work on a self-hosted install and rather than investigate further I think we should just move ahead and deprecate (to avoid confusion) as has been the plan anyway.

Yes we can and I think we just confirmed what I thought:

> Looking at the how altinity/clickhouse-operator works, I doubt this value in our repo had ever worked as expected.

If this never worked, I'm 👍 to also don't make this a breaking change (as it never worked before)

File: charts/posthog/templates/clickhouse_instance.yaml [hunk 1]
```diff
@@ -110,11 +110,6 @@ spec:
                   mountPath: /var/lib/clickhouse
               {{- end }}
 
-              {{- if .Values.clickhouse.useNodeSelector }}
-              nodeSelector:
```
Comment (hazzadous) [inline charts/posthog/templates/clickhouse_instance.yaml:114]: yeah looks like this should be at the pod spec context and not the container context? As per https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/

**Review by hazzadous — APPROVED** [review]
Looks good, as for update it seems there's no harm in making this a non-breaking change but we should obviously have somewhere in the release notes that they should use `nodeSelector` instead.

**Comment by hazzadous** [comment]
> As we usually write release notes for breaking changes only, what do you think about simply updating the docs?

Yeah I think that’s ok, if a user is having issues with it hopefully they’ll do some searches and see that it’s deprecated from GitHub issues and that it never worked, from the pr

**Comment by guidoiaquinti** [comment]
We can also add a deprecation notice here in Helm. Good suggestion

**Review by hazzadous — APPROVED** [review]

**Comment by tiina303** [comment]
how about we update the already existing release note about this https://posthog.com/docs/self-host/deploy/upgrade-notes#upgrading-from-9xx

File: charts/posthog/values.yaml [hunk 2]
```diff
@@ -552,9 +552,6 @@ clickhouse:
   # -- Service Type: LoadBalancer (allows external access) or NodePort (more secure, no extra cost)
   serviceType: NodePort
 
-  # -- If enabled, operator will prefer k8s nodes with tag `clickhouse:true`
-  useNodeSelector: false
```
Comment (tiina303) [inline charts/posthog/values.yaml:556]: can we get `nodeSelector` documented in values it currently isn't (i.e. isn't in ALL_VALUES)

**Review by tiina303 — APPROVED** [review]

**Pull request merged**